dlp/deid_test.py

```diff
@@ -0,0 +1,168 @@
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is *********' in out
+
+
+def test_deidentify_with_mask_ignore_insensitive_data(capsys):
+    harmless_string = 'My favorite color is blue'
+
+    deid.deidentify_with_mask(parent, harmless_string)
+
+    out, _ = capsys.readouterr()
+    assert harmless_string in out
+
+
+def test_deidentify_with_mask_masking_character_specified(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string, masking_character='#')
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is #########' in out
+
+
+def test_deidentify_with_mask_masking_number_specified(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string, number_to_mask=7)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is *******27' in out
+
+
+def test_deidentify_with_mask_handles_masking_number_error(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string, number_to_mask=-3)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is *********' in out
+# [END deidentify_with_mask]
+
+
+# [START Deidentify with FPE]
+def test_deidentify_with_fpe(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_fpe(parent, harmful_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is' in out
+    assert '372819127' not in out
+
+
+def test_deidentify_with_fpe_uses_surrogate_info_types(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_fpe(parent, harmful_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name,
+                             surrogate_type=surrogate_type)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is SSN_TOKEN' in out
+    assert '372819127' not in out
+
+
+def test_deidentify_with_fpe_ignores_insensitive_data(capsys):
+    harmless_string = 'My favorite color is blue'
+
+    deid.deidentify_with_fpe(parent, harmless_string, alphabet='NUMERIC',
+                             wrapped_key=wrapped_key, key_name=key_name)
+
+    out, _ = capsys.readouterr()
+    assert harmless_string in out
+# [END Deidentify with FPE]
+
+
+# [START Deidentify with date shift]
+def test_deidentify_with_date_shift(capsys):
+    deid.deidentify_with_date_shift(parent, input_csv_file=csv_file,
+                                    output_csv_file=output_csv_file,
+                                    lower_bound_days=date_shifted_amount,
+                                    upper_bound_days=date_shifted_amount,
+                                    date_fields=date_fields)
+
+    out, _ = capsys.readouterr()
+
+    assert 'Successful' in out
+    # read in csv??
```

Comment:
No need to read in the CSV at this time, but go ahead and remove this comment.

Reply:
done